megamek/src/megamek/client/bot/princess/BasicPathRanker.java

```diff
@@ -1110,89 +1099,81 @@ private double calcWaterHazard(Entity movingUnit, Hex hex, MoveStep step, MovePa
                 submergedLocations.add(loc);
             }
         }
-        logMsg.append("\n\t\tSubmerged locations: ")
-                .append(submergedLocations.size());
+        logger.trace("Submerged locations: {}", submergedLocations);
 
         int hazardValue = 0;
         for (int loc : submergedLocations) {
-            logMsg.append("\n\t\t\tLocation ").append(loc).append(" is ");
-
             // Only locations without armor can breach in movement phase.
             if (movingUnit.getArmor(loc) > 0) {
-                logMsg.append(" not breached (0).");
+                logger.trace("Location {} is not breached (0).", loc);
                 continue;
             }
 
             // Meks or ProtoMeks having a head or torso breach is deadly.
             // For other units, any breach is deadly.
             // noinspection ConstantConditions
-            if (Mek.LOC_HEAD == loc ||
-                    Mek.LOC_CT == loc ||
-                    ProtoMek.LOC_HEAD == loc ||
-                    ProtoMek.LOC_TORSO == loc ||
-                    (!(movingUnit instanceof Mek) &&
-                            !(movingUnit instanceof ProtoMek))) {
-                logMsg.append(" breached and critical (1000).");
+            if ((Mek.LOC_HEAD == loc) ||
+                (Mek.LOC_CT == loc) ||
+                (ProtoMek.LOC_HEAD == loc) ||
+                (ProtoMek.LOC_TORSO == loc) ||
+                !(movingUnit.isMek() && !(movingUnit.isProtoMek()))
```

Comment:
I believe the first `!` must go inside the parentheses; this new construction will evaluate to `true` if the unit is a ProtoMek, but previously it would only evaluate to `true` if the unit is not a Mek _and_ not a ProtoMek.